charmhelpers/contrib/openstack/templating.py

```diff
@@ -124,6 +127,20 @@ def complete_contexts(self):
         self.context()
         return self._complete_contexts
 
+    @property
```

Comment:
I can remove that but this is just for visual appearance - by looking at this I do not need to grep in different codebases to know that this might be used outside the class.

Reply:
I tend to think this pattern just adds noise to the code - it'll be simpler without it and you can avoid writing a docstring!